docs/concepts/policy/pod-security-policy.md

```diff
@@ -1,97 +1,371 @@
-control the security features a pod has access to. These settings fall
-into three categories:
-
-- *Controlled by a Boolean*: Fields of this type default to the most
-restrictive value.
-- *Controlled by an allowable set*: Fields of this type are checked
-against the set to ensure their values are allowed.
-- *Controlled by a strategy*: Items that have a strategy to provide
-a mechanism to generate the value and a mechanism to ensure that a
-specified value falls into the set of allowable values.
-
-
-## Strategies
-
-### RunAsUser
+| Control Aspect                                      | Field Names                                 |
+| ----------------------------------------------------| ------------------------------------------- |
+| Running of privileged containers                    | `privileged`                                |
+| Usage of the root namespaces                        | [`hostPID`, `hostIPC`](#host-namespaces)    |
+| Usage of host networking and ports                  | [`hostNetwork`, `hostPorts`](#host-namespaces) |
+| Usage of volume types                               | [`volumes`](#volumes-and-file-systems)      |
+| Usage of the host filesystem                        | [`allowedHostPaths`](#volumes-and-file-systems) |
+| Allocating an FSGroup that owns the pod's volumes   | [`fsGroup`](#volumes-and-file-systems)      |
+| Requiring the use of a read only root file system   | [`readOnlyRootFilesystem`](#volumes-and-file-systems) |
+| The user and group IDs of the container             | [`runAsUser`, `supplementalGroups`](#users-and-groups) |
+| Restricting root privileges                         | [`allowPrivilegeEscalation`, `defaultAllowPrivilegeEscalation`](#privilege-escalation) |
+| Linux capabilities                                  | [`defaultAddCapabilities`, `requiredDropCapabilities`, `allowedCapabilities`](#capabilities) |
+| The SELinux context of the container                | [`seLinux`](#selinux)                       |
+| The AppArmor profile used by containers             | [annotations](#apparmor)                    |
+| The seccomp profile used by containers              | [annotations](#seccomp)                     |
 
-- *MustRunAs* - Requires a `range` to be configured. Uses the first value
-of the range as the default. Validates against the configured range.
-- *MustRunAsNonRoot* - Requires that the pod be submitted with a non-zero
-`runAsUser` or have the `USER` directive defined in the image. No default
-provided.
-- *RunAsAny* - No default provided. Allows any `runAsUser` to be specified.
 
-### SELinux
+## Enabling Pod Security Policies
 
-- *MustRunAs* - Requires `seLinuxOptions` to be configured if not using
-pre-allocated values. Uses `seLinuxOptions` as the default. Validates against
-`seLinuxOptions`.
-- *RunAsAny* - No default provided. Allows any `seLinuxOptions` to be
-specified.
+Pod security policy control is implemented as an optional [admission
```

Comment:
I'm a bit surprised that admission plugin is optional. How the control will work without it?

Reply:
I'm not sure I understand your question. If the plugin is not enabled, then PodSecurityPolicy won't be enforced, but the cluster will still work.

Reply:
So, it means, that there will be no control and all pods will be accepted.

I'd rather remove "optional" word from here because in reality nothing will works as we expected without admission plugin.

Reply:
I think the phrasing is more that it is optional but highly encouraged for production environments.  You don't **have** to run it, you just have to accept the risk.

Reply:
It's not enabled by default, it's in beta, and until recently wasn't tested in any e2e environments. I don't think we can claim it's not optional, but I added a note that it's recommended.